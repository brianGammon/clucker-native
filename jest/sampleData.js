const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
