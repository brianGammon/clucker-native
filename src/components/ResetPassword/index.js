import ResetPasswordComponent from './ResetPassword';

export default ResetPasswordComponent;
