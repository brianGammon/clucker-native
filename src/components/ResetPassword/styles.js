import { StyleSheet } from 'react-native';

const baseText = {
  padding: 20,
};
export default StyleSheet.create({
  success: {
    ...baseText,
    color: 'green',
  },
  message: {
    ...baseText,
    color: 'grey',
  },
});
