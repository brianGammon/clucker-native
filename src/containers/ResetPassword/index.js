import React from 'react';
import { Button, View } from 'react-native';

interface Props {
  navigation: any;
}

export default class ResetPassword extends React.Component<Props> {
  static navigationOptions = {
    title: 'Reset Password',
  };

  render() {
    const { navigation } = this.props;
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Button onPress={() => navigation.goBack()} title="Go Back" />
      </View>
    );
  }
}
