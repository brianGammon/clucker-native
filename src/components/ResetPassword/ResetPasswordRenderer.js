import * as React from 'react';
import { FieldGroup, FieldControl } from 'react-reactive-form';
import { View, Button, Text } from 'react-native';
import FormInput from '../FormInput';
import styles from './styles';

type Props = {
  navigation: any,
  form: any,
  error: string,
  successMessage: string,
  handleSubmit: () => void,
};

const ResetPasswordRenderer = ({
  navigation,
  form,
  error,
  successMessage,
  handleSubmit,
}: Props) => (
  // <View style={styles.container}>
  //   {error && <Text>{error}</Text>}
  //   {successMessage && !error && <Text>{successMessage}</Text>}
  //   <Text>Reset your password</Text>
  //   <View style={styles.formGroup}>
  //     <Text style={styles.label}>Email:</Text>
  //     <TextInput
  //       style={styles.input}
  //       value={email}
  //       autoCapitalize="none"
  //       onChangeText={handleChangeText}
  //       autoFocus
  //     />
  //   </View>
  //   <Button onPress={handleSendPasswordResetEmail} title="Send Email" />
  //   <Button onPress={() => navigation.goBack()} title="Back To Sign In" />
  // </View>
  <View style={styles.container}>
    <View style={styles.signInContainer}>
      <View>
        <Text style={styles.title}>Clucker</Text>
      </View>
      {error && <Text style={styles.error}>{error}</Text>}
      {successMessage
        && !error && <Text style={styles.success}>{successMessage}</Text>}
      <FieldGroup
        control={form}
        render={({ invalid }) => (
          <View style={styles.formContainer}>
            <FieldControl
              name="email"
              render={FormInput}
              meta={{ label: 'Email Addess', autoCapitalize: 'none' }}
            />

            <Button
              disabled={invalid}
              onPress={handleSubmit}
              title="Send Email"
            />
            <Button
              onPress={() => navigation.goBack()}
              title="Back To Sign In"
            />
          </View>
        )}
      />
    </View>
  </View>
);

export default ResetPasswordRenderer;
