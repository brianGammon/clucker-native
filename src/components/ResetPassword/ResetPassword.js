import * as React from 'react';
import { connect } from 'react-redux';
import { FormBuilder, Validators } from 'react-reactive-form';
import { actionTypes } from '../../redux/constants';
import { resetPasswordRequested } from '../../redux/actions';
import ResetPasswordRenderer from './ResetPasswordRenderer';

type Props = {
  navigation: any,
  error: string,
  inProgress: boolean,
  sendPasswordResetEmail: (email: string) => void,
  clearError: () => void,
};

type State = {
  successMessage: string,
};

class ResetPassword extends React.Component<Props, State> {
  static navigationOptions = {
    title: 'Reset Password',
  };

  form = FormBuilder.group({
    email: ['', [Validators.email, Validators.required]],
  });

  state = { successMessage: null };

  componentDidUpdate(prevProps) {
    const { inProgress: prevInProgress } = prevProps;
    const { inProgress, error } = this.props;
    if (prevInProgress && !inProgress && !error) {
      this.setSuccess();
    }
  }

  componentWillUnmount() {
    const { error, clearError } = this.props;
    if (error) {
      clearError();
    }
  }

  setSuccess = () => {
    this.setState({ successMessage: 'Check your inbox for an email.' });
  };

  handleSubmit = () => {
    const { email } = this.form.value;
    const { sendPasswordResetEmail } = this.props;
    sendPasswordResetEmail(email);
  };

  render() {
    const { navigation, error } = this.props;
    const { successMessage } = this.state;
    return (
      <ResetPasswordRenderer
        navigation={navigation}
        form={this.form}
        successMessage={successMessage}
        error={error}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

const mapStateToProps = ({ auth: { errors, inProgress } }) => ({
  error: errors.resetPassword,
  inProgress,
});

const mapDispatchToProps = dispatch => ({
  sendPasswordResetEmail: (email: string) => dispatch(resetPasswordRequested(email)),
  clearError: () => dispatch({ type: actionTypes.CLEAR_AUTH_ERROR }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ResetPassword);
