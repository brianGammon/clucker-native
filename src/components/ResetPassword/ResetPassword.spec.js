import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/no-flocks.json';
import ResetPassword from '.';
import ResetPasswordRenderer from './ResetPasswordRenderer';

describe('ResetPassword component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);
  test('Should render', () => {
    const wrapper = shallow(<ResetPassword store={store} />).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('ResetPasswordRenderer renders when no error or success message', () => {
    const wrapper = shallow(
      <ResetPasswordRenderer
        email="test@exmaple.com"
        error=""
        successMessage=""
        handleSendPasswordResetEmail={() => {}}
        handleChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('ResetPasswordRenderer renders when success message, no error', () => {
    const wrapper = shallow(
      <ResetPasswordRenderer
        email="test@exmaple.com"
        error=""
        successMessage="Check your inbox for an email."
        handleSendPasswordResetEmail={() => {}}
        handleChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('ResetPasswordRenderer renders with error message, no success message', () => {
    const wrapper = shallow(
      <ResetPasswordRenderer
        email="test@exmaple.com"
        error="Some error occurred"
        successMessage=""
        handleSendPasswordResetEmail={() => {}}
        handleChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('ResetPasswordRenderer renders with both error and success message', () => {
    const wrapper = shallow(
      <ResetPasswordRenderer
        email="test@exmaple.com"
        error="Some error occurred"
        successMessage="Check your inbox for an email."
        handleSendPasswordResetEmail={() => {}}
        handleChangeText={() => {}}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
