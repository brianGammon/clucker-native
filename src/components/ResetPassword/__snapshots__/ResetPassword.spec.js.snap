// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword component: ResetPasswordRenderer renders when no error or success message 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .85)",
        "borderRadius": 20,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        Clucker
      </Component>
    </Component>
    <FieldGroup
      render={[Function]}
      strict={true}
    />
  </Component>
</Component>
`;

exports[`ResetPassword component: ResetPasswordRenderer renders when success message, no error 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .85)",
        "borderRadius": 20,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        Clucker
      </Component>
    </Component>
    <Component
      style={
        Object {
          "color": "green",
          "padding": 20,
        }
      }
    >
      Check your inbox for an email.
    </Component>
    <FieldGroup
      render={[Function]}
      strict={true}
    />
  </Component>
</Component>
`;

exports[`ResetPassword component: ResetPasswordRenderer renders with both error and success message 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .85)",
        "borderRadius": 20,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        Clucker
      </Component>
    </Component>
    <Component
      style={
        Object {
          "color": "red",
          "padding": 20,
        }
      }
    >
      Some error occurred
    </Component>
    <FieldGroup
      render={[Function]}
      strict={true}
    />
  </Component>
</Component>
`;

exports[`ResetPassword component: ResetPasswordRenderer renders with error message, no success message 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, .85)",
        "borderRadius": 20,
        "padding": 20,
        "width": "90%",
      }
    }
  >
    <Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 40,
            "fontWeight": "bold",
          }
        }
      >
        Clucker
      </Component>
    </Component>
    <Component
      style={
        Object {
          "color": "red",
          "padding": 20,
        }
      }
    >
      Some error occurred
    </Component>
    <FieldGroup
      render={[Function]}
      strict={true}
    />
  </Component>
</Component>
`;

exports[`ResetPassword component: Should render 1`] = `
<ResetPasswordRenderer
  error={null}
  form={
    FormGroup {
      "_onDisabledChange": Array [],
      "_pendingChange": false,
      "_pendingDirty": false,
      "_pendingTouched": false,
      "asyncValidator": null,
      "controls": Object {
        "email": FormControl {
          "_onCollectionChange": [Function],
          "_onDisabledChange": Array [],
          "_parent": [Circular],
          "_pendingChange": true,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "_pendingValue": "",
          "active": false,
          "asyncValidator": null,
          "errors": Object {
            "required": true,
          },
          "formState": "",
          "get": [Function],
          "getError": [Function],
          "handler": [Function],
          "hasError": [Function],
          "meta": Object {},
          "onBlur": [Function],
          "onBlurChanges": Observable {
            "observers": Array [],
          },
          "onChange": [Function],
          "onFocus": [Function],
          "onValueChanges": Observable {
            "observers": Array [],
          },
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "INVALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": [Function],
          "validatorsOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": Array [
              [Function],
              [Function],
            ],
          },
          "value": "",
          "valueChanges": Observable {
            "observers": Array [],
          },
        },
      },
      "errors": null,
      "get": [Function],
      "getError": [Function],
      "handleSubmit": [Function],
      "hasError": [Function],
      "meta": Object {},
      "patchValue": [Function],
      "pristine": true,
      "reset": [Function],
      "setValue": [Function],
      "stateChanges": Observable {
        "observers": Array [],
      },
      "status": "INVALID",
      "statusChanges": Observable {
        "observers": Array [],
      },
      "submitted": false,
      "touched": false,
      "validator": null,
      "validatorOrOpts": Object {
        "asyncValidators": null,
        "updateOn": null,
        "validators": null,
      },
      "value": Object {
        "email": "",
      },
      "valueChanges": Observable {
        "observers": Array [],
      },
    }
  }
  handleSubmit={[Function]}
  successMessage={null}
/>
`;
